src/code/z_kankyo.c

```diff
@@ -832,9 +2717,267 @@ u8 sSandstormEnvColors[] = {
+                                  sSandstormPrimColors[sNextSandstormColorIndex * 3 + index + 1], sSandstormLerpScale);
+            primColor.b =
+                (s32)F32_LERP_ALT(sSandstormPrimColors[sSandstormColorIndex * 3 + index + 2],
+                                  sSandstormPrimColors[sNextSandstormColorIndex * 3 + index + 2], sSandstormLerpScale);
+            envColor.r =
+                (s32)F32_LERP_ALT(sSandstormEnvColors[sSandstormColorIndex * 3 + index + 0],
+                                  sSandstormEnvColors[sNextSandstormColorIndex * 3 + index + 0], sSandstormLerpScale);
+            envColor.g =
+                (s32)F32_LERP_ALT(sSandstormEnvColors[sSandstormColorIndex * 3 + index + 1],
+                                  sSandstormEnvColors[sNextSandstormColorIndex * 3 + index + 1], sSandstormLerpScale);
+            envColor.b =
+                (s32)F32_LERP_ALT(sSandstormEnvColors[sSandstormColorIndex * 3 + index + 2],
+                                  sSandstormEnvColors[sNextSandstormColorIndex * 3 + index + 2], sSandstormLerpScale);
+        }
+
+        envColor.r = ((envColor.r * sp98) + ((6.0f - sp98) * primColor.r)) * (1.0f / 6.0f);
+        envColor.g = ((envColor.g * sp98) + ((6.0f - sp98) * primColor.g)) * (1.0f / 6.0f);
+        envColor.b = ((envColor.b * sp98) + ((6.0f - sp98) * primColor.b)) * (1.0f / 6.0f);
+
+        sp96 = (sSandstormScroll * (11.0f / 6.0f));
+        sp94 = (sSandstormScroll * (9.0f / 6.0f));
+        sp92 = (sSandstormScroll * (6.0f / 6.0f));
+
+        OPEN_DISPS(play->state.gfxCtx);
+
+        POLY_XLU_DISP = Gfx_SetupDL64(POLY_XLU_DISP);
+
+        gDPSetAlphaDither(POLY_XLU_DISP++, G_AD_NOISE);
+        gDPSetColorDither(POLY_XLU_DISP++, G_CD_NOISE);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, primColor.r, primColor.g, primColor.b, play->envCtx.sandstormPrimA);
+        gDPSetEnvColor(POLY_XLU_DISP++, envColor.r, envColor.g, envColor.b, play->envCtx.sandstormEnvA);
+        gSPSegment(POLY_XLU_DISP++, 0x08,
+                   Gfx_TwoTexScroll(play->state.gfxCtx, G_TX_RENDERTILE, (u32)sp96 % 4096, 0, 512, 32, 1,
+                                    (u32)sp94 % 4096, 4095 - ((u32)sp92 % 4096), 256, 64));
+        gDPSetTextureLUT(POLY_XLU_DISP++, G_TT_NONE);
+        gSPDisplayList(POLY_XLU_DISP++, gFieldSandstormDL);
+
+        CLOSE_DISPS(play->state.gfxCtx);
+    }
+
+    sSandstormScroll += (s32)sp98;
+}
+
+s32 Environment_AdjustLights(PlayState* play, f32 arg1, f32 arg2, f32 arg3, f32 arg4) {
+    f32 temp;
+    s32 i;
+    Player* player = GET_PLAYER(play);
+
+    if (play->roomCtx.curRoom.behaviorType1 == ROOM_BEHAVIOR_TYPE1_5) {
+        return 0;
+    }
+
+    if (!Play_CamIsNotFixed(&play->state)) {
+        return 0;
+    }
+    if (play->unk_18880) {
+        return 0;
+    }
+    if ((player != NULL) && (player->stateFlags1 & PLAYER_STATE1_2)) {
+        return 0;
+    }
```

Comment:
```c
overlays/actors/ovl_En_Elf/z_en_elf.c:    if ((this->unk_240 >= 0.0f) && Environment_AdjustLights(play, SQ(this->unk_240) * this->unk_240,
```
Hmm, elf is currently the only place that calls this that uses the return and it uses it as a bool, should these be true/false?